nameko/__init__.py

```diff
@@ -1,8 +1,71 @@
 from copy import deepcopy
+from six import wraps
+from six.moves import UserDict
 
 
-config = {}
+class Config(UserDict):
+
+    def setup(self, config):
+        self.data.clear()
+        self.data.update(config)
+
+    def patch(self, context_config, clear=False):
+        """
+        A context manager and decorator for updating config just for the given context
+        or decorated function execution.
+
+        Intended to be used mainly in tests:
+
+            @pytest.yield_fixture
+            def memory_rabbit_config():
+                with nameko.config.patch({"AMQP_URI": "memory://"}):
+                    yield
```

Comment:
It's a little confusing that this fixture is defined but not used. I think you can drop this and just demonstrate the decorator usage.

Reply:
:+1: